### PR TITLE
New members API for members page

### LINKED ISSUE #262: GET new members who recently signed up

### PR DESCRIPTION
**Pull request by rajakvk**
Created new PR with new branch

Closes #262 

This PR should only be merged after successful run/merge of this PR #351

### CODE REVIEW

File: controllers/members.js
Comment (swarajpure) [inline controllers/members.js:35]: This is to find the idle members only from old members, right? Just confirming.

File: models/members.js
Comment (swarajpure) [inline models/members.js:38]: I get what is `new` and `blocked` are meant to do, but what is `member` doing? Didn't get that, can you please elaborate a little bit?

File: test/integration/members.test.js
Comment (swarajpure) [inline test/integration/members.test.js:50]: `newMembers` would also be having `isMember` true, right?

**Review by swarajpure — COMMENTED** [review]
Having some minor queries, please check.

File: controllers/members.js
Comment (rajakvk) [reply controllers/members.js:35]: Basically the structure of the returned json has been changed from allMembers = [{},{},{},...] to allMembers = {newMembers: [{},{},{},...], oldMembers: [{},{},{},...]}.

So I believe, we are considering idle members from oldMembers not from newMembers. That is my take. 

We should confirm this assumption with AO. @sumitd94 

In fact, Sumit suggested this json structure change in his previous review.

File: models/members.js
Comment (rajakvk) [reply models/members.js:38]: TDLR; member case included to handle future scenario, not necessary right now.

IsMember is existing field. userType newly introduced field along with this PR. We need to discuss with AO, @sumitd94 
1. what to do with these two inter-related fields?
2. are we going to keep both the fields?
3. are we going to remove isMember field?
4. If we remove isMember, how to deal with related code changes?
5. If we remove isMemebr, how to deal with existing data?

File: test/integration/members.test.js
Comment (rajakvk) [reply test/integration/members.test.js:50]: Right now 'Yes'. 
All are members, plus their userType value would be 'new'.

Comment (swarajpure) [reply test/integration/members.test.js:50]: Can we assert on that as well?

File: controllers/members.js
Comment (swarajpure) [reply controllers/members.js:35]: I'm also assuming the same. @sumitd94 Please confirm.

Comment (sumitd94) [reply controllers/members.js:35]: Yes, idle members should be returned from the list of `old-members` (here by old-members we are referring to all those users who has `isMember` set to `true`

File: test/integration/members.test.js
Comment (rajakvk) [reply test/integration/members.test.js:50]: This needs discussion based on my questions raised above.

File: models/members.js
Comment (sumitd94) [reply models/members.js:38]: What do you suggest here?

Comment (sumitd94) [reply models/members.js:38]: According to me

1. For now let's keep both, userType will contain the values like `new, blocked, member`. here the `member` is same as `isMember = true`
2. For now yes
3. In future yes
4. Need to think, open to suggestion
5. Need to think, open to suggestion

Comment (swarajpure) [reply models/members.js:38]: I still didn't get why do we need `member` in `userType` if we already have `isMember` 😅

Comment (rajakvk) [reply models/members.js:38]: @swarajpure, if we decide to keep isMember field forever then 'member' in 'userType' is not necessary. But that decision is not yet taken.

Comment (swarajpure) [reply models/members.js:38]: @sumitd94 please unblock @rajakvk

File: models/members.js
Comment (sumitd94) [inline models/members.js:34]: I think we can re-structure this to something like this

```javascript
if (memberData.roles && memberData.roles.member) {
allmembers.oldMembers.push(curatedMemberData)
} else {
...new members
}
```

**Review by sumitd94 — CHANGES_REQUESTED** [review]
One small change, please.

File: models/members.js
Comment (rajakvk) [reply models/members.js]: Done with the update.

**Comment by rajakvk** [comment]
Integration test failing when executing /members/idle related tests due to changes in 'tasks' end point changes. I would suggest a discussion needed to proceed further. Also @AO take a call on how to deal with inter-related test cases. @ankurnarkhede @sumitd94 @ankushdharkar

**Comment by ankurnarkhede** [comment]
Will get the PR https://github.com/Real-Dev-Squad/website-backend/pull/296 merged ASAP to resolve the failing CI, and thus merge this.

**Comment by ankurnarkhede** [comment]
@rajakvk , is this PR blocked on any other PR? As per I can see, the CI is passing for develop. The tests for this branch are throwing 500, I suppose this could be because of an incorrectly handled exception. 🤔

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]
@rajakvk  , is this PR blocked on any other PR? As per I can see, the CI is passing for develop. The tests for this branch are throwing 500, I suppose this could be because of an incorrectly handled exception. 🤔

File: models/members.js
Comment (harshith-venkatesh) [inline models/members.js:31]: I believe this should be oldMembers

**Comment by ankurnarkhede** [comment]
As discussed, can we simply add `isNewMember = <boolean>` (or any other nice name) in the user response object. This will also maintain the existing contract between API and UI and thus won't break things.

**Review by sumitd94 — DISMISSED** [review]
Looks perfect, Thank you so much 🥇

File: models/members.js
Comment (ankushdharkar) [inline models/members.js:34]: Nit:
```
          allMembers.isMember = !!(memberData.roles && memberData.roles.member)
```

File: models/members.js
Comment (ankushdharkar) [inline models/members.js:14]: Should this be called `fetchUsers` now? Given that all members are users, but all users are not members

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: models/members.js
Comment (sumitd94) [reply models/members.js:34]: shouldn't this be `curatedMemberData.isMember`?

**Review by sumitd94 — APPROVED** [review]

**Pull request merged**
